consensus/src/experimental/buffer_manager.rs

```diff
@@ -175,31 +133,21 @@ impl StateManager {
         ordered_blocks: OrderedBlocks,
     ) -> anyhow::Result<()> {
         let OrderedBlocks {
-            blocks,
+            ordered_blocks,
             ordered_proof,
             callback,
         } = ordered_blocks;
 
         // push blocks to buffer
-        for eb in blocks.iter() {
-            self.buffer
-                .push_back(BufferItem::Block(Arc::new(eb.clone())));
-        }
-
-        self.buffer.push_back(BufferItem::Proof(Box::new(ProofItem(
-            ordered_proof.clone(),
-            BTreeMap::new(),
+        self.buffer.push_back(BufferItem::new_ordered(
```

Comment:
this function can direclty take a OrderedBlock?

Reply:
We need the `ordered_blocks` field in sending the request. Probably it's good to destruct it here.